src/tls.c

```diff
@@ -740,7 +740,7 @@ static int connTLSConnect(connection *conn_, const char *addr, int port, const c
     return C_OK;
 }
 
-static int connTLSWrite(connection *conn_, const void *data, size_t data_len) {
+static ssize_t connTLSWrite(connection *conn_, const void *data, size_t data_len) {
```

Comment:
SSL_write takes an int.. 
i suppose we better clamp the input rather than truncate it (possibly leading to negative value)

Reply:
Then I'll revert this change first, as it doesn't cause a bug.

Reply:
@oranagra I think ```SSL_read``` and ```SSL_write```  should be upgraded to ```SSL_read_ex``` and ```SSL_write_ex```

Reply:
@sundb i'm arguing that we should lave the entire set of `conn` changes for another PR.
this PR is already too complicated to follow, and as i stated earlier, connRead will never actually return more than a few megabytes (that are already present in the OS socket buffers), so there's no real bug in the new code we introduce in networking.c

Reply:
@oranagra Sorry for the misunderstanding, done it.